### PR TITLE
feat(footer-core): Add component - TWIG-73

### PR DESCRIPTION
**Pull request by papegaill**
# PR description

Please drop a few lines about the PR: what it does, how to test it, etc.

## QA Checklist

In order to ensure a safe and quick review, please check that your PR follow those guidelines:

- [ ] I have put the vanilla component as `devDependencies`
- [ ] I have put the specs package as `devDependencies`
- [ ] I have added the components directly used in the twig file (with `include` or `embed`) as `dependencies`
- [ ] My component is listed in `@ecl-twig/ec-components`'s `dependencies`
- [ ] My variables naming follow the guidelines (snake case for twig)
- [ ] I have provided tests
- [ ] I have provided documentation (for the "notes" tab)
- [ ] If my local `yarn.lock` contains changes, I have committed it
- [ ] I have given my PR the proper label (`pr: review needed` to indicate that I'm done and now waiting for a review ,`pr: wip` to indicate that I'm actively working on it ...)

### CODE REVIEW

File: src/ec/packages/ec-component-footer-core/footer-core.html.twig [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    {% if section.section_class_name is defined and section.section_class_name is not empty %}
+      {% set _section_class = _section_class ~ ' ' ~ section.section_class_name %}
+    {% endif %}
+    <div class="{{ _section_class }}">
+      {%- if section.title is defined and section.title is not empty -%}
+        {% include '../ec-component-link/link.html.twig' with {
+          link: section.title|default({})|merge({
+            type: 'standalone',
+          }),
+          extra_classes: 'ecl-footer-core__title'
```
Comment (planctus) [inline src/ec/packages/ec-component-footer-core/footer-core.html.twig]: there should be an href for the link to be valid, and i don't see it, in ecl we have an hash, an issue with the specs data?

File: src/ec/packages/ec-component-footer-core/footer-core.html.twig [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  {% endfor %}
+{% endif %}
+
+{# Print the result #}
+
+<footer class="{{ _css_class }}"{{ _extra_attributes|raw }}>
+  <div class="ecl-container ecl-footer-core__container">
+  {% for key, section in _sections %}
+    {% set _section_id = key + 1 %}
+    {% set _section_class = 'ecl-footer-core__section ecl-footer-core__section' ~ _section_id %}
```
[the remark this answers is not in the document]
Comment (papegaill) [reply src/ec/packages/ec-component-footer-core/footer-core.html.twig]: True :)

File: src/ec/packages/ec-component-footer-core/footer-core.html.twig [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+      {% set _extra_attributes = _extra_attributes ~ ' ' ~ attr.name %}
+    {% endif %}
+  {% endfor %}
+{% endif %}
+
+{# Print the result #}
+
+<footer class="{{ _css_class }}"{{ _extra_attributes|raw }}>
+  <div class="ecl-container ecl-footer-core__container">
+  {% for key, section in _sections %}
```
[the remark this answers is not in the document]
Comment (papegaill) [reply src/ec/packages/ec-component-footer-core/footer-core.html.twig]: done, thx.

File: src/ec/packages/ec-component-footer-core/footer-core.html.twig (hunk 1 above)
Comment (papegaill) [reply src/ec/packages/ec-component-footer-core/footer-core.html.twig]: I have used the formatlink function to adapt title data coming form the specs. I also reviewed the all the links structure for the notes doc component.

File: src/ec/packages/ec-component-footer-core/footer-core.html.twig [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+          {% for item in section.links %}
+            <li class="ecl-footer-core__list-item">
+              {% include '../ec-component-link/link.html.twig' with item|merge({
+                link: item.link|default({})|merge({
+                  type: 'standalone',
+                }),
+                icon: item.icon|default({})|merge({
+                  size: 'xs',
+                }),
+                extra_classes: 'ecl-footer__section-link'
```
[the remark this answers is not in the document]
Comment (papegaill) [reply src/ec/packages/ec-component-footer-core/footer-core.html.twig]: Updated, thx.

File: src/ec/packages/ec-component-footer-core/footer-core.story.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { storiesOf } from '@storybook/html';
+import { withKnobs } from '@storybook/addon-knobs';
```
Comment (planctus) [inline src/ec/packages/ec-component-footer-core/footer-core.story.js]: I think we can avoid importing the knobs as well as setting them in the decorator, we don't have knobs defined for this component, right?

File: src/ec/packages/ec-component-footer-core/footer-core.html.twig [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+            {% for item in section.links %}
+              <li class="ecl-footer-core__list-item">
+                {% include '../ec-component-link/link.html.twig' with item|merge({
+                  link: item.link|default({})|merge({
+                    type: 'standalone',
+                  }),
+                  icon: item.icon|default({})|merge({
+                    size: 'xs',
+                  }),
+                  extra_classes: 'ecl-footer__core__link'
```
[the remark this answers is not in the document]
Comment (papegaill) [reply src/ec/packages/ec-component-footer-core/footer-core.html.twig]: oups, updated !

File: src/ec/packages/ec-component-footer-core/footer-core.story.js (hunk 5 above)
Comment (papegaill) [reply src/ec/packages/ec-component-footer-core/footer-core.story.js]: removed

**Pull request merged**